libs/server/Resp/AdminCommands.cs

```diff
@@ -488,6 +557,79 @@ private bool ProcessAdminCommands<TGarnetApi>(RespCommand command, ReadOnlySpan<
             return true;
         }
 
+
+        void ProcessHelloCommand(int? protocolVersion, ReadOnlySpan<byte> username, ReadOnlySpan<byte> password, string clientName)
+        {
+            if (protocolVersion != null)
+            {
+                if (protocolVersion.Value is < 2 or > 3)
+                {
+                    while (!RespWriteUtils.WriteError("ERR Unsupported protocol version"u8, ref dcurr, dend))
+                        SendAndReset();
+                    return;
+                }
+
+                this.protocolVersion = protocolVersion.Value;
+            }
+
+            if (username != default)
+            {
+                if (!this.AuthenticateUser(username, password))
+                {
+                    if (username.IsEmpty)
+                    {
+                        while (!RespWriteUtils.WriteError("WRONGPASS Invalid password"u8, ref dcurr, dend))
+                            SendAndReset();
+                        return;
+                    }
+                    else
+                    {
+                        while (!RespWriteUtils.WriteError("WRONGPASS Invalid username/password combination"u8, ref dcurr, dend))
+                            SendAndReset();
+                        return;
+                    }
+                }
+            }
+
+            if (clientName != null)
+            {
+                this.clientName = clientName;
+            }
+
+            (string, string)[] helloResult =
+                [
+                    ("server", "redis"),
+                    ("version", storeWrapper.redisProtocolVersion),
+                    ("garnet_version", storeWrapper.version),
+                    ("proto", $"{this.protocolVersion}"),
+                    ("id", "63"),
```

Comment:
Shouldn't this return a client ID?

Reply:
We don't support a client ID concept, so I just return a constant here.

Reply:
Shouldn't we return something like -1 or 0 then? Might be less confusing

Reply:
just followed another server's model where it returned 64. can revisit in a polishing PR later.